langchain/document_loaders/audio.py

```diff
@@ -0,0 +1,35 @@
+import os
+import openai
+from typing import List, Iterator
+from langchain.docstore.document import Document
+from langchain.document_loaders.generic import GenericLoader
+
+class AudioFileLoader(GenericLoader):
+    """Document loader for audio files using audio-to-text transcription with OpenAI Whisper model.
+    
+    Example:
+
+    .. code-block:: python
+        from langchain.document_loaders import AudioFileLoader
+        audio_file_path = "/path/to/directory"
+        loader = AudioFileLoader(audio_file_path)
+        loader.load()
+    """
+
+    def __init__(self, audio_file_path: str):
```

Comment:
Let's avoid this kind of overloading of the __init__ with any specific audio file path since that makes it impossible to use the classmethods for the generic loader to pick up files by pattern.

----

I suspect we might need to update how the parser is being specified if we want to be able to decouple loader from parsing more -- we likely we need to lift the parser to be a class level attribute

Reply:
> In principle, we don't even need to introduce a new loader since the GenericLoader can handle this.

right, this makes sense. the duplication of logic between loader and parser was odd. let me test this.